src/Apps/Order/Routes/__tests__/Shipping.jest.tsx

```diff
@@ -199,21 +205,135 @@ const testMe: ShippingTestQuery$rawResponse["me"] = {
+        me={props.me}
+        // @ts-ignore
+        isCommittingMutation={isCommittingMutation}
+      />
+    </MockBoot>
+  ),
+  query: graphql`
+    query ShippingTestQuery @raw_response_type @relay_test_operation {
+      order: commerceOrder(id: "unused") {
+        ...Shipping_order
+      }
+      me {
+        ...Shipping_me
+      }
+    }
+  `,
+}
+
+const { renderWithRelay } = setupTestWrapperTL<ShippingTestQuery>(
+  testWrapperConfig
+)
+describe("Shipping [@testing-library/react]", () => {
+  const validAddress: Address = {
+    name: "Joelle Van Dyne",
+    country: "US",
+    postalCode: "10013",
+    addressLine1: "401 Broadway",
+    addressLine2: "Suite 25",
+    city: "New York",
+    region: "NY",
+    phoneNumber: "422-424-4242",
+  }
+  describe("US address verification", () => {
+    const addressVerificationResult = {
+      __typename: "VerifyAddressType",
+      verificationStatus: "NOT_FOUND",
+      suggestedAddresses: [],
+      inputAddress: {
+        lines: ["401 Broadway", "Suite 25", "New York, NY 10013", "USA"],
+        address: validAddress,
+      },
+    }
+    beforeAll(() => {
+      ;(useFeatureFlag as jest.Mock).mockImplementation(
+        (featureName: string) => featureName === "address_verification_us"
+      )
+    })
+
+    beforeEach(() => {
+      isCommittingMutation = false
+    })
+
+    afterEach(() => {
+      jest.clearAllMocks()
+      relayEnv.mockClear()
+    })
+
+    afterAll(() => {
+      ;(useFeatureFlag as jest.Mock).mockReset()
+    })
+
+    it("triggers basic form validation first", async () => {
+      const mockResolvers = {
+        CommerceOrder: () => testOrder,
+        Me: () => emptyTestMe,
+      }
+      const { env } = renderWithRelay(mockResolvers, true, {}, relayEnv)
+
+      env.mock.resolveMostRecentOperation(operation => {
+        return MockPayloadGenerator.generate(operation, mockResolvers)
+      })
+      const submitButton = screen.getByText("Save and Continue")
+      submitButton.click()
+
+      await waitFor(() => {
+        const errorFields = screen.getAllByText("This field is required")
+        expect(errorFields).toHaveLength(7)
+      })
+
+      TLHelpers.fillAddressFields(TLHelpers.validAddress)
+      await waitFor(() => {
+        expect(
+          screen.queryByText("This field is required")
+        ).not.toBeInTheDocument()
+      })
+
+      submitButton.click()
+
+      // Resolve the address verification query
+      await waitFor(() => {
+        act(() => {
+          env.mock.resolveMostRecentOperation(operation => {
+            // TODO: this doesn't work. WHY?
+            // return MockPayloadGenerator.generate(operation, {
+            //   VerifyAddressMutationType: () => addressVerificationResult,
+            // })
+            return {
+              data: {
+                verifyAddress: {
+                  verifyAddressOrError: addressVerificationResult,
+                },
+              },
+            }
```

Comment:
After finally getting the query renderer wrapping the Address Verification Flow to use my mocked env I still couldn't get the payload generator to respect my mock resolver, so i just passed data explicitly :/